var/spack/repos/builtin/packages/genie/package.py

```diff
@@ -0,0 +1,265 @@
+    # Disables this check.
+    patch("genie_disable_gopt_with_compiler_check.patch", level=0, when="@2.11:")
+
+    # Flags for GENIE"s optional but disabled by default features
+    variant(
+        "vleextension",
+        default=False,
+        description="Enable GENIE very low energy (1 MeV - 100 MeV) extension",
+    )
+    variant(
+        "validationtools",
+        default=False,
+        description="GENIE physics model validation tools",
+    )
+    variant("test", default=False, description="Enable test programs")
+    variant("t2k", default=False, description="Enable T2K-specific generation app")
+    variant("fnal", default=False, description="Enable T2K-specific generation app")
+    variant(
+        "atmo",
+        default=False,
+        description="Enable GENIE Atmospheric neutrino event generation app",
+    )
+    variant(
+        "nucleondecay",
+        default=False,
+        description="Enable GENIE Nucleon decay event generation app",
+    )
+    variant(
+        "masterclass",
+        default=False,
+        description="Enable GENIE neutrino masterclass app",
+    )
+
+    conflicts(
+        "+validationtools",
+        when="@2.8.6:2.8.6",
+        msg="Validation tools in GENIE 2.8.6 don't build.",
+    )
+
+    phases = ["configure", "build", "install"]
+
+    def setup_build_environment(self, env):
+        env.set("GENIE", self.stage.source_path)
+        return super(Genie, self).setup_build_environment(env)
+
+    def setup_run_environment(self, env):
+        env.set("GENIE", self.prefix)
+        return super(Genie, self).setup_run_environment(env)
+
+    def configure(self, spec, prefix):
+        configure = Executable("./configure")
+        args = self._configure_args(spec, prefix)
+        configure(*args)
+
+    def build(self, spec, prefix):
+        # parallel build is not supported on GENIE 2
+        self._make(parallel=spec.satisfies("@3:"))
+
+    def install(self, spec, prefix):
+        # GENIE make install does not support parallel jobs
+        self._make("install", parallel=False)
+        # GENIE requires these files to be present at runtime, but doesn"t install them
+        # so we must install them ourselves
+        # install_tree function is injected into scope by spack build_environment.py
+        install_tree("config", os.sep.join((prefix, "config")))
+        install_tree("data", os.sep.join((prefix, "data")))
+
+    def _configure_args(self, spec, prefix):
+        args = [
+            "--prefix=%s" % prefix,
+            "--with-compiler=%s" % os.environ["CC"],
+            "--with-libxml2-inc=%s%slibxml2" % (spec["libxml2"].prefix.include, os.sep),
+            "--with-libxml2-lib=%s" % spec["libxml2"].prefix.lib,
+            "--with-log4cpp-inc=%s" % spec["log4cpp"].prefix.include,
+            "--with-log4cpp-lib=%s" % spec["log4cpp"].prefix.lib,
+            "--with-pythia6-lib=%s" % spec["pythia6"].prefix.lib,
```

Comment:
Use Python3-style positional arguments `'...={0}'.format(prefix)` or in most cases here simply `'...=' + prefix` will do

Reply:
Thanks. Implemented in 240eb22.